client/src/modules/stock/lots/registry.js

```diff
@@ -107,53 +131,91 @@ function StockLotsController(Stock, Notify,
   // clear all filters
   function clearFilters() {
     SearchFilterFormat.clearFilters(reload);
-  }
-
-  // load stock lots in the grid
-  function load(filters) {
-    var today = { defaultPeriod: 'today' };
-    var params = filters;
+  }  
 
-    var noFilter = (!filters);
-    var noAttributes = (noFilter || (Object.keys(filters).length === 0));
+  // initialize module
+  function startup() {
+    load(Stock.lotFilters.formatHTTP(true));
+    vm.latestViewFilters = Stock.lotFilters.formatView();
+  }
 
-    if (noAttributes) {
-      params = today;
-      vm.isToday = true;
-      vm.filters = { display: today, identifiers: today };
-      vm.formatedFilters = SearchFilterFormat.formatDisplayNames(vm.filters.display);
-    }
+  /**
+   * @function errorHandler
+   *
+   * @description
+   * Uses Notify to show an error in case the server sends back an information.
+   * Triggers the error state on the grid.
+   */
+  function errorHandler(error) {
+    vm.hasError = true;
+    Notify.handleError(error);
+  }
 
-    vm.loading = true;
+  /**
+   * @function toggleLoadingIndicator
+   *
+   * @description
+   * Toggles the grid's loading indicator to eliminate the flash when rendering
+   * lots movements and allow a better UX for slow loads.
+   */
+  function toggleLoadingIndicator() {
+    vm.loading = !vm.loading;
+  }
 
-    Stock.lots.read(null, params).then(function (lots) {
-      vm.loading = false;
+  // load stock lots in the grid
+  function load(filters) {
+    vm.hasError = false;
+    toggleLoadingIndicator();
 
+    Stock.lots.read(null, filters)
+    .then(function(lots){
       vm.gridOptions.data = lots;
-
       vm.grouping.unfoldAllGroups();
     })
-    .catch(Notify.handleError);
+    .catch(errorHandler)
+    .finally(function (){
+      toggleLoadingIndicator();     
+    });
   }
 
-  // search modal
-  function search() {
-    Modal.openSearchLots()
-    .then(function (filters) {
-      if (!filters) { return; }
+  // remove a filter with from the filter object, save the filters and reload
+  vm.onRemoveFilter = function onRemoveFilter(key) {
+    Stock.removeLotFilter(key);
 
-      vm.isToday = false;
-      reload(filters);
-    })
-    .catch(Notify.handleError);
+    Stock.cacheLotFilters();
+    vm.latestViewFilters = Stock.lotFilters.formatView();
+
+    return load(Stock.lotFilters.formatHTTP(true));
   }
 
-  // reload
-  function reload(filters) {
-    vm.filters = filters;
-    vm.formatedFilters = SearchFilterFormat.formatDisplayNames(filters.display);
-    load(filters.identifiers);
+  function search() {
+    var filtersSnapshot = Stock.lotFilters.formatHTTP();
+
+    Modal.openSearchLots(filtersSnapshot)
+      .then(function (changes) {
+        Stock.lotFilters.replaceFilters(changes);
+        Stock.cacheLotFilters();
+        vm.latestViewFilters = Stock.lotFilters.formatView();
+
+        return load(Stock.lotFilters.formatHTTP(true));
+      })
+      .catch(angular.noop);
```

Comment:
This should probably be handled... but I don't think it will ever get called since `load()` has its own error handling.